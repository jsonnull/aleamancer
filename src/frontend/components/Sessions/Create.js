// @flow
import React from 'react'
import Button from 'frontend/components/Button'

type Props = {
  createSession: Function
}

const CreateButton = Button.extend`
  display: block;
  flex: 0;
  margin-bottom: auto;
  margin-right: auto;
`

const Create = (props: Props) => {
  return <CreateButton onClick={props.createSession}>Create Game</CreateButton>
}

export default Create
