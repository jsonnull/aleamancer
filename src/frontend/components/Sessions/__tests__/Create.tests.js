// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import Create from '../Create.js'

describe('Sessions Create component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Create createSession={() => {}} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
