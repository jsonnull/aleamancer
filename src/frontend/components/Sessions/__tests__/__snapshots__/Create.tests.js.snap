// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sessions Create component renders correctly 1`] = `
.c0 {
  display: inline-block;
  font-size: 1.3rem;
  line-height: 3rem;
  height: 3rem;
  padding: 0 1rem;
  outline: 0;
  color: #eee;
  background: #046380;
  box-shadow: inset 0 -20px 10px -10px #045d78;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition: background,box-shadow;
  transition: background,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: block;
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  margin-bottom: auto;
  margin-right: auto;
}

.c0:hover {
  background: #045a74;
  box-shadow: inset 0 -20px 10px -10px #03546d;
  -webkit-transition: background,box-shadow;
  transition: background,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
}

<button
  className="c0"
  onClick={[Function]}
>
  Create Game
</button>
`;
